caproto/threading/client.py

```diff
@@ -1789,6 +1792,110 @@ def __del__(self):
             pass
 
 
+class Batch:
+    """
+    Accumulate requests and then issue them all in batch.
+
+    Parameters
+    ----------
+    timeout : number or None
+        Number of seconds to wait before ignoring late responses. Default
+        is 2.
```

Comment:
Maybe in this case a linear function, proportional to number of requests, would be reasonable?

Reply:
Better do this on all the methods then. Punting to https://github.com/NSLS-II/caproto/issues/284 (but plan to address today)